src/Hacking/netscriptCanHack.ts

```diff
@@ -9,28 +9,28 @@ import { IReturnStatus } from "../types";
 import { Player } from "@player";
 import { Server } from "../Server/Server";
 
-function baseCheck(server: Server, fnName: string): IReturnStatus {
+function baseCheck(server: Server, actionName: string): IReturnStatus {
   const hostname = server.hostname;
 
-  if (!("requiredHackingSkill" in server)) {
+  if (server.purchasedByPlayer) {
```

Comment:
Just to double-check: There are no cases where we are loosening the check, right? The message claims the requiredHackingSkill check was only for hacknet, but that doesn't mean it's right.

Reply:
The old check is useless. Before we call the triplet `netscriptCan-X`, we always check `!(server instanceof Server)` (the type of the first parameter of `baseCheck` is `Server`). Hacknet Node and Hacknet Server are not `Server`.